raitools/raitools/raianalyzer/raianalyzer.py

```diff
@@ -0,0 +1,144 @@
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param train: The training dataset.
+        :type train: pandas.DataFrame
+        :param test: The test dataset.
+        :type test: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: The task to run, can be `classification` or
+            `regression`.
+        :type task_type: str
+        """
+        self.model = model
+        self.train = train
+        self.test = test
+        self.target_column = target_column
+        self.task_type = task_type
+        self._causal_manager = CausalManager()
+        self._counterfactual_manager = CounterfactualManager()
+        self._error_analysis_manager = ErrorAnalysisManager()
+        self._classes = train[target_column].unique()
+        self._explainer_manager = ExplainerManager(model, train, test,
+                                                   target_column,
+                                                   self._classes)
+        self._fairness_manager = FairnessManager()
+        self._managers = [self._causal_manager,
+                          self._counterfactual_manager,
+                          self._error_analysis_manager,
+                          self._explainer_manager,
+                          self._fairness_manager]
+
+    @property
+    def causal(self):
+        """Get the causal manager.
+
+        :return: The causal manager.
+        :rtype: CausalManager
+        """
+        return self._causal_manager
+
+    @property
+    def counterfactual(self):
+        """Get the counterfactual manager.
+
+        :return: The counterfactual manager.
+        :rtype: CounterfactualManager
+        """
+        return self._counterfactual_manager
+
+    @property
+    def erroranalysis(self):
+        """Get the error analysis manager.
+
+        :return: The error analysis manager.
+        :rtype: ErrorAnalysisManager
+        """
+        return self._error_analysis_manager
+
+    @property
+    def explainer(self):
+        """Get the explainer manager.
+
+        :return: The explainer manager.
+        :rtype: ExplainerManager
+        """
+        return self._explainer_manager
+
+    @property
+    def fairness(self):
+        """Get the fairness manager.
+
+        :return: The fairness manager.
+        :rtype: FairnessManager
+        """
+        return self._fairness_manager
+
+    def compute(self):
+        """Calls compute on each of the managers."""
+        for manager in self._managers:
+            manager.compute()
+
+    def list(self):
+        """List information about each of the managers.
+
+        :return: Information about each of the managers.
+        :rtype: dict
+        """
+        configs = {}
+        for manager in self._managers:
+            configs[manager.name] = manager.list()
+
+    def save(self, path):
+        # TODO: implement
```

Comment:
Same comment about `UnimplementedError` here

Reply:
done, I also raised it in the managers, except for list and compute for now, since top level calls those on all managers